api/src/main/java/com/redhat/insights/http/InsightsFileWritingClient.java

```diff
@@ -46,15 +46,13 @@ public void decorate(InsightsReport report) {
   @Override
   public void sendInsightsReport(String filename, InsightsReport report) {
     decorate(report);
-    String reportJson = report.serialize();
-    //    logger.debug(reportJson);
 
     // Can't reuse upload path - as this may be called as part of fallback
     Path p = Paths.get(config.getArchiveUploadDir(), filename + ".json");
     try {
       Files.write(
```

Comment:
From discussions with @ehsavoie I recall that the report has to be amended on fallbacks between clients, due to the API design and the need to patch a top-level report entry,

Reply:
Perfect, so we can just leave the improvement as it is, at this point: I would like to avoid using the File API later one and just make it stream into it by Jackson (which would be the best). see https://github.com/RedHatInsights/insights-java-client/pull/110#issuecomment-1611742763 for more info